CMakeLists.txt

```diff
@@ -346,39 +350,45 @@ install(FILES ${CMAKE_BINARY_DIR}/etc/gitinfo.txt DESTINATION ${CMAKE_INSTALL_SY
 
 #---Recurse into the given subdirectories.  This does not actually cause another cmake executable
 #  to run. The same process will walk through the project's entire directory structure.
-add_subdirectory (core)
-add_subdirectory (math)
-add_subdirectory (hist)
-add_subdirectory (tree)
-add_subdirectory (io)
-add_subdirectory (net)
-add_subdirectory (graf2d)
-add_subdirectory (graf3d)
-add_subdirectory (gui)
-if(html)
-  add_subdirectory (html)
-endif()
-add_subdirectory (montecarlo)
-if(geom)
-  add_subdirectory (geom)
-endif()
-add_subdirectory (rootx)
-add_subdirectory (misc)
-add_subdirectory (main)
-add_subdirectory (bindings)
-add_subdirectory (sql)
-if(proof)
-  add_subdirectory(proof)
-endif()
-if(tmva)
-  add_subdirectory(tmva)
-endif()
-if(roofit)
-  add_subdirectory(roofit)
-endif()
-
-ROOT_ADD_TEST_SUBDIRECTORY(test)
-ROOT_ADD_TEST_SUBDIRECTORY(tutorials)
+if(NOT ROOT_ENABLE_PROJECTS)
+       set(ROOT_ENABLE_PROJECTS "All")
+endif()
+
+if(ROOT_ENABLE_PROJECTS STREQUAL "All" OR ROOT_ENABLE_PROJECTS STREQUAL "Essentials")
+       add_subdirectory (core)
```

Comment:
Can you implement `root_add_subdirectory`, change `add_subdirectory` to it and do all of the heavy lifting and conditionals in there?

Reply:
ok, I did the changes